csv2ir.py

```diff
@@ -1,39 +1,73 @@
+#!/usr/bin/python3
 import argparse
 import os
 import sys
 import csv
 
 
-def convert(csv_in, ir_out, protocol="NECext"):
+
+SUPPORTED_IR = [
+    "NEC", "NECext", "NEC42", "NEC42ext",
+    "RC5", "RC5X", "RC6",
+    "Samsung32",
+    "SIRC", "SIRC15", "SIRC20",
+]
```

Comment:
Rather than having this list, and then a separate `get_protocol()` with a nested `if`, what if we just have a dictionary like:
```
SUPPORTED_IR = {"Sony12" : "SIRC", "Sony15" : "SIRC15", "Sony20" : "SIRC20", "Tivo unit=0" : "NECext", "NECx1" : "NECext", "NECx2" : "NECext", "NEC2-f16" : "NECext", "RC5-7F" : "RC5X"}
```
and then do something like this below:
```
ir_protocol = next(csv_reader)[1]
if ir_protocol in SUPPORTED_IR:
    ir_protocol = SUPPORTED_IR[ir_protocol]
else:
    print("file {} uses IR protocol {}: not supported, skipped".format(csv_in, ir_protocol), file=sys.stderr)
    return
```
Fewer things to maintain that way?

Reply:
Done although used two dictionaries,  one for mapping, one for validating support (unless you prefer the map dictionary to map values to themselves)

Reply:
Ah, I didn't catch that some of the protocols don't need to be mapped per se. In that case, this solution makes sense to me.